graphql2/graphqlapp/app.go

```diff
@@ -85,31 +85,30 @@ type App struct {
 	FormatDestFunc func(context.Context, notification.DestType, string) string
 }
 
-func mustAuth(h http.Handler) http.Handler {
-	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
-		err := permission.LimitCheckAny(req.Context())
-		if errutil.HTTPError(req.Context(), w, err) {
-			return
-		}
-
-		h.ServeHTTP(w, req)
-	})
-}
-
-func (a *App) PlayHandler() http.Handler {
+func (a *App) PlayHandler(w http.ResponseWriter, req *http.Request) {
 	var data struct {
-		Version     string
-		PackageName string
+		ApplicationName string
+		Version         string
+		PackageName     string
 	}
+
+	err := permission.LimitCheckAny(req.Context())
+	if err != nil {
+		http.Error(w, http.StatusText(http.StatusUnauthorized), http.StatusUnauthorized)
+		return
+	}
+
+	cfg := config.FromContext(req.Context())
+
+	data.ApplicationName = cfg.ApplicationName()
 	data.Version = playVersion
 	data.PackageName = playPackageName
-	return mustAuth(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
-		err := playTmpl.Execute(w, data)
-		if err != nil {
-			log.Log(req.Context(), err)
-			http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
-		}
-	}))
+
+	err = playTmpl.Execute(w, data)
+	if err != nil {
+		log.Log(req.Context(), err)
```

Comment:
use `errutil.HTTPError` for this and the one above so we have one place we're handling status codes & logging for HTTP errors